cookbooks/aws-parallelcluster-test/recipes/tests.rb

```diff
@@ -99,6 +99,9 @@
       command "ls #{node['cluster']['slurm']['install_dir']}"
       user node['cluster']['cluster_user']
     end
+    execute 'check memory cgroup' do
+      command "[[ $(grep memory /proc/cgroups | awk '{print $4}') == 1 ]]"
```

Comment:
The memory constraining mechanism in Slurm (used when `ConstrainRAMSpace=yes`) is based on cgroups and requires that the memory cgroup controller be enabled. This can be checked via the `/proc/cgroups` file, which contains a table formatted in this way:
```
$ column -t /proc/cgroups 
#subsys_name  hierarchy  num_cgroups  enabled
cpuset        4          1            1
cpu           9          1            1
...
memory        2          200          1
```

We need to check that the line with `memory` is there, and that the fourth field is a `1`.

Reply:
I'd add the expected output we're checking in a comment in the code itself